### PR TITLE
raft: implemented leader lease when quorum check is on

### PR DESCRIPTION
**Pull request by tiancaiamao**

### CODE REVIEW

File: src/raftstore/store/config.rs [hunk 1]
```diff
@@ -52,6 +52,8 @@ pub struct Config {
     // When entry count exceed this value, gc will be forced trigger.
     pub raft_log_gc_limit: u64,
 
+    pub check_quorum: bool,
```
Comment (siddontang) [inline src/raftstore/store/config.rs]: raft_check_quorum

File: tests/raftstore/test_transfer_leader.rs [hunk 2]
```diff
@@ -18,6 +18,11 @@ use super::server::new_server_cluster;
 use std::time::Duration;
 
 fn test_transfer_leader<T: Simulator>(cluster: &mut Cluster<T>) {
+    // TODO transfer leader conflicts with leader lease feature, which
+    // is enabled when check_quorum on.
+    // After etcd solve it, we'll port and may remove this line then.
```
Comment (siddontang) [inline tests/raftstore/test_transfer_leader.rs]: solves

File: tests/raftstore/test_transport.rs [hunk 3]
```diff
@@ -14,9 +14,14 @@
 use super::cluster::{Cluster, Simulator};
 use super::node::new_node_cluster;
 use super::server::new_server_cluster;
-use super::util::{must_get_equal, new_peer};
+use super::util::{must_get_equal, new_peer, sleep_ms};
 
 fn test_partition_write<T: Simulator>(cluster: &mut Cluster<T>) {
+    // TODO transfer leader conflicts with leader lease feature, which
+    // is enabled when check_quorum on.
+    // After etcd solve it, we'll port and may remove this line then.
```
Comment (siddontang) [inline tests/raftstore/test_transport.rs]: solves

File: tests/raftstore/test_transport.rs [hunk 4]
```diff
@@ -38,6 +43,10 @@ fn test_partition_write<T: Simulator>(cluster: &mut Cluster<T>) {
 
     // leader in minority, new leader should be elected
     cluster.partition(vec![1, 2], vec![3, 4, 5]);
+    // leader lease make it take longer to elect the new leader
```
Comment (siddontang) [inline tests/raftstore/test_transport.rs]: makes

**Comment by tiancaiamao** [comment]
this feature conflict with transfer leader currently.
maybe re-add later

**Pull request closed without merging**